dda-cunh/webserv
# Feature/cgi basic implementation

First CGI implementation

Implemented features to test:

- Execute CGI based on certain file extension. Executes .cgi file.
- Works with POST, GET, DELETE methods
- Make the route able to accept uploaded files 
- use the full path as PATH_INFO
- server should work with one CGI

`
 curl -X POST -F 'file=@a.txt' http://localhost:4242/cgi/file_handler.cgi
 curl  GET  http://localhost:4242/cgi/file_handler.cgi
 curl  -X DELETE -F 'file=@a.txt' http://localhost:4242/cgi/file_handler.cgi
 curl  GET  http://localhost:4242/cgi/file_handler.cgi
`

still to implement:
-  configure where files should be saved 
- Output of the CGI. If no content_length is returned from the CGI, EOF will mark the end of the returned data.
- for chunked request, your server needs to unchunk ut, the CGI will expect EOF as end of the body??
- program should call the CGI with the file requested as first argument??
- The CGI should be run in the correct directory for relative path file access??
